app/controllers/group_assignment_invitations_controller.rb

```diff
@@ -1,70 +1,64 @@
   before_action :check_user_not_group_member,       only: [:show]
 
   before_action :authorize_group_access, only: [:accept_invitation]
 
-  before_action :ensure_github_repo_exists, only: [:successful_invitation]
-
   def show
-    @groups = invitation.groups.map { |group| [group.title, group.id] }
+    @groups = current_invitation.groups.map { |group| [group.title, group.id] }
   end
 
   def accept; end
 
   def accept_assignment
-    create_group_assignment_repo { redirect_to successful_invitation_group_assignment_invitation_path }
+    create_submission do
+      redirect_to success_group_assignment_invitation_path
+    end
   end
 
   def accept_invitation
     selected_group       = Group.find_by(id: group_params[:id])
     selected_group_title = group_params[:title]
 
-    create_group_assignment_repo(selected_group: selected_group,
-                                 new_group_title: selected_group_title) do
-      redirect_to successful_invitation_group_assignment_invitation_path
+    options = {
+      selected_group: selected_group,
+      new_group_title: selected_group_title
+    }
+
+    create_submission(options) do
+      redirect_to success_group_assignment_invitation_path
     end
   end
 
-  def successful_invitation; end
-
   private
 
-  def required_scopes
-    GitHubClassroom::Scopes::GROUP_ASSIGNMENT_STUDENT
-  end
-
   def authorize_group_access
     group_id = group_params[:id]
 
     return if group_id.blank?
     group = Group.find(group_id)
     validate_max_members_not_exceeded!(group)
-    return if group_assignment.grouping.groups.find_by(id: group_id)
+    return if current_assignment.grouping.groups.find_by(id: group_id)
 
     raise NotAuthorized, 'You are not permitted to select this team'
   end
 
-  def validate_max_members_not_exceeded!(group)
-    return unless group.present? && group_assignment.present? && group_assignment.max_members.present?
-    return unless group.repo_accesses.count >= group_assignment.max_members
-    raise NotAuthorized, "This team has reached its maximum member limit of #{group_assignment.max_members}."
+  def check_group_not_previous_acceptee
+    return unless group.present? && current_submission.present?
+    redirect_to success_group_assignment_invitation_path
   end
 
-  def group
-    repo_access = current_user.repo_accesses.find_by(organization: organization)
-    return unless repo_access.present? && repo_access.groups.present?
-
-    @group ||= repo_access.groups.find_by(grouping: group_assignment.grouping)
+  def check_user_not_group_member
+    return if group.blank?
+    redirect_to accept_group_assignment_invitation_path
   end
-  helper_method :group
 
-  def create_group_assignment_repo(selected_group: group, new_group_title: nil)
-    users_group_assignment_repo = invitation.redeem_for(current_user, selected_group, new_group_title)
+  def create_submission(selected_group: group, new_group_title: nil)
```

Comment:
Any way we could re-word this? We're introducing the concept of assignment submissions with the deadline stuff, so I could see this getting a little confusing.